villain/interp.rkt

```diff
@@ -100,7 +100,19 @@
           [(Wild)  (interp-env e r ds)]
           [(Var x) (interp-env e (ext r x v) ds)]
           [(Lit l)
-           (if (eq? l v)
+           (if (and (number? l) (number? v))
```

Comment:
I think it would be better to expand out a mach on `l` so that you have code specific to each kind of literal, i.e. numbers, should use `=`, everything else (I think) should use `eq?`.